care/emr/api/viewsets/totp.py

```diff
@@ -0,0 +1,334 @@
+
+        user.totp_secret = encrypted_secret
+        user.save(update_fields=["totp_secret"])
+
+        response_data = TOTPSetupResponse(uri=uri, secret_key=secret)
+        return Response(response_data.model_dump())
+
+    @staticmethod
+    def _generate_backup_codes(count: int = 10) -> list[str]:
+        """Generate 8-digit backup codes."""
+        codes = []
+        for _ in range(count):
+            code = "".join(choice(digits) for _ in range(8))
+            codes.append(code)
+        return codes
+
+    @extend_schema(
+        description="Verify TOTP code and enable 2FA",
+        request=TOTPVerifyRequest,
+        responses={
+            200: TOTPVerifyResponse,
+            400: {"type": "object", "properties": {"error": {"type": "string"}}},
+        },
+    )
+    @action(detail=False, methods=["POST"])
+    def verify(self, request):
+        request_data = TOTPVerifyRequest(**request.data)
+        user = request.user
+
+        if not user.totp_secret:
+            return Response(
+                {"error": "TOTP not configured"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        secret = decrypt_string(user.totp_secret)
+        totp = TOTP(secret)
+
+        if totp.verify(request_data.code):
+            backup_codes = self._generate_backup_codes()
+
+            mfa_settings = user.mfa_settings or {}
+            mfa_settings["totp"] = {
+                "enabled": True,
+                "enabled_at": timezone.now().isoformat(),
+                "backup_codes": [
+                    {
+                        "code": make_password(code),
+                        "used": False,
+                        "created_at": timezone.now().isoformat(),
+                    }
+                    for code in backup_codes
+                ],
+            }
+            user.mfa_settings = mfa_settings
+            user.save(update_fields=["mfa_settings"])
+
+            send_totp_enabled_email.delay(user.email, user.username)
+
+            response_data = TOTPVerifyResponse(
+                message="Two-factor authentication has been enabled successfully. Please save your backup codes in a secure location.",
+                backup_codes=backup_codes,
+            )
+            return Response(response_data.model_dump())
+
+        return Response({"error": "Invalid code"}, status=status.HTTP_400_BAD_REQUEST)
+
+    @extend_schema(
+        description="Verify TOTP code during login",
+        request=TOTPLoginRequest,
+        responses={
+            200: TOTPLoginResponse,
+            400: {"type": "object", "properties": {"error": {"type": "string"}}},
+        },
+    )
+    @action(
+        detail=False,
+        methods=["POST"],
+        permission_classes=[],
+        authentication_classes=[],
+    )
+    def login(self, request):
+        request_data = TOTPLoginRequest(**request.data)
+
+        token = RefreshToken(request_data.temp_token)
```

Comment:
Need exception handling around this, if an invalid token is given it will raise an error